tracing/client.go

```diff
@@ -203,45 +212,45 @@ func newClient(ctx context.Context, conf *config.Config, token string) (*Client,
 				clientConn, _ := grpc.NewClient(grpcAddress, dialOps...)
 				streamClient, err := tempo.NewgRPCClient(clientConn)
 				if err != nil {
-					log.Errorf("Error creating gRPC Tempo Client %s", err.Error())
+					zl.Error().Msgf("Error creating gRPC Tempo Client: %v", err)
 					return nil, nil
 				}
-				return &Client{httpTracingClient: httpTracingClient, grpcClient: streamClient, httpClient: client, baseURL: u, ctx: ctx}, nil
+				return &Client{httpTracingClient: httpTracingClient, grpcClient: streamClient, httpClient: client, baseURL: u, clientCtx: clientCtx}, nil
 			}
 		} else {
 			httpTracingClient, err = jaeger.NewJaegerClient(client, u)
 			if err != nil {
 				return nil, err
 			}
 		}
-		return &Client{httpTracingClient: httpTracingClient, httpClient: client, baseURL: u, ctx: ctx}, nil
+		return &Client{httpTracingClient: httpTracingClient, httpClient: client, baseURL: u, clientCtx: clientCtx}, nil
 	}
 }
 
 // GetAppTraces fetches traces of an app
-func (in *Client) GetAppTraces(namespace, app string, q models.TracingQuery) (*model.TracingResponse, error) {
+func (in *Client) GetAppTraces(ctx context.Context, namespace, app string, q models.TracingQuery) (*model.TracingResponse, error) {
 	serviceName := BuildTracingServiceName(namespace, app)
 	if in.grpcClient == nil {
-		return in.httpTracingClient.GetAppTracesHTTP(in.httpClient, in.baseURL, serviceName, q)
+		return in.httpTracingClient.GetAppTracesHTTP(ctx, in.httpClient, in.baseURL, serviceName, q)
 	}
-	return in.grpcClient.FindTraces(in.ctx, serviceName, q)
+	return in.grpcClient.FindTraces(ctx, in.clientCtx, serviceName, q)
```

Comment:
This would be simpler if we didn't save the context on the struct. The rub here is in trying to combine two loggers right? idk if there's an easy way to do that with zerolog but I'd rather we pass one context and have some kind of `getLogger(ctx)` method that pulls the logger from the context and adds the tracing group each time we want a logger in the `Client` instead of having to pass two contexts. Context is also used for cancellation, timeout, etc. Which one do you use for that and which one do you use for the logger? IMO it's confusing.